src/ws_distance.F90

```diff
@@ -0,0 +1,229 @@
+!-*- mode: F90 -*-!
+!                                                            !
+! Copyright (C) 2016 Lorenzo Paulatto                        !
+!                                                            !
+! This file is distributed under the terms of the GNU        !
+! General Public License. See the file `LICENSE' in          !
+! the root directory of the present distribution, or         !
+! http://www.gnu.org/copyleft/gpl.txt .                      !
+!                                                            !
+!------------------------------------------------------------!
+
+module w90_ws_distance
+!     if( allocated( wdist_ndeg ) ) then
+!        deallocate( wdist_ndeg, stat=ierr  )
+!        if (ierr/=0) call io_error('Error in deallocating wdist_ndeg in param_dealloc')
+!     end if
+!     if( allocated( wdist_shiftj_wsi ) ) then
+!        deallocate( wdist_shiftj_wsi, stat=ierr  )
+!        if (ierr/=0) call io_error('Error in deallocating wdist_shiftj_wsi in param_dealloc')
+!     end if
+
+  use w90_constants, only : dp
+
+  implicit none
+
+  private
+  !
+  public :: ws_translate_dist, clean_ws_translate
+  !
+  ! The number of unit cells to shift WF j to put its centre inside the Wigner-Seitz
+  ! of wannier function i. If several shifts are equivalent (i.e. they take the function
+  ! on the edge of the WS) they are all listed
+  integer, public, save, allocatable :: wdist_shiftj_wsi(:,:,:,:,:)!(3,ndegenx,num_wann,num_wann,nrpts)
+  ! The number of equivalent shifts (see above)
+  integer, public, save, allocatable :: wdist_ndeg(:,:,:)!(num_wann,num_wann,nrpts)
+  !
+  logical, save, public :: use_ws_distance = .true.
```

Comment:
Ok,
I'm also changing a sign according to a suggestion by Marco Gibertini,
although I was not able to find a case where it actually makes a difference
(it would need to be very low symmetry..)

Can you tell me how to update the pull request? Or do I have to close it
and make another one?

On 14 September 2016 at 19:18, Giovanni Pizzi <notifications@github.com>
wrote:

> In src/ws_distance.F90
> <https://github.com/wannier-developers/wannier90/pull/9#discussion_r78793124>
> :
>
> > +  use w90_constants, only : dp
> > +
> > +  implicit none
> > +
> > +  private
> > +  !
> > +  public :: ws_translate_dist, clean_ws_translate
> > +  !
> > +  ! The number of unit cells to shift WF j to put its centre inside the Wigner-Seitz
> > +  ! of wannier function i. If several shifts are equivalent (i.e. they take the function
> > +  ! on the edge of the WS) they are all listed
> > +  integer, public, save, allocatable :: wdist_shiftj_wsi(:,:,:,:,:)!(3,ndegenx,num_wann,num_wann,nrpts)
> > +  ! The number of equivalent shifts (see above)
> > +  integer, public, save, allocatable :: wdist_ndeg(:,:,:)!(num_wann,num_wann,nrpts)
> > +  !
> > +  logical, save, public :: use_ws_distance = .true.
>
> Can you add this as an input parameter, or at least set it by default to
> .false. so we can merge and do a few more tests locally before making
> this the default behaviour? Thanks!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wannier-developers/wannier90/pull/9/files/ed5892553fe99075ec489529a965afbe02131047#r78793124>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABNXpKHO4MWVIHe3VR1Lc9gg0jMp7FEkks5qqCxJgaJpZM4J8g76>
> .
>



-- 
Lorenzo Paulatto - Paris